packages/cli-platform-ios/src/commands/buildIOS/index.ts

```diff
@@ -0,0 +1,244 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import path from 'path';
+import chalk from 'chalk';
+import {Config} from '@react-native-community/cli-types';
+import {
+  logger,
+  CLIError,
+  getDefaultUserTerminal,
+} from '@react-native-community/cli-tools';
+import {Device} from '../../types';
+import {BuildFlags, buildProject} from './buildProject';
+import {getDestinationSimulator} from '../../tools/getDestinationSimulator';
+import {getDevices} from '../../tools/getDevices';
+
+export interface FlagsT extends BuildFlags {
+  configuration?: string;
+  simulator?: string;
+  device?: string | true;
+  udid?: string;
+  scheme?: string;
+}
+
+function buildIOS(_: Array<string>, ctx: Config, args: FlagsT) {
+  if (!ctx.project.ios) {
+    throw new CLIError(
+      'iOS project folder not found. Are you sure this is a React Native project?',
+    );
+  }
+
+  if (args.configuration) {
+    logger.warn('--configuration has been deprecated. Use --mode instead.');
+  }
```

Comment:
since it's deprecated, maybe it's worth assigning the value of `args.configuration` to `args.mode`? Otherwise the build will run with default mode, and if it's not what the dev expected, it will be necessary to run the command again. WDYT?

Reply:
Extended warning and assigned config to mode.